### PR TITLE
Fixes schema propagation while creating a hydrator pipeline

### PR DESCRIPTION
**Pull request by ajainarayanan**
#### FIXME
  - ~~Right now the schema propagation doesn't work for implicit schemas. The reason being, the first node have to be opened up to fetch the ui config from node proxy to get the implicit schema. And this happens only when we connect all the nodes and just click on schema icons on the edges without opening any node (I am assuming that if people have already used a plugin, say Twitter, and they know that it has an implicit schema and expect that to propagate automatically without even opening the node).~~
- We fetch the properties and node config for a node as soon as its added to the canvas. So now we have this information and whenever we encounter an implicit schema down the line we stop the propagation there for that particular path.

- We need to re-word the confirmation message we show while propagating the schema.

### CODE REVIEW

**Comment by ajainarayanan** [comment]
propagation12
propagation1

File: cdap-ui/app/features/hydrator/controllers/create/partials/nodeconfig-ctrl.js [hunk 1]
```diff
@@ -77,9 +78,25 @@ class NodeConfigController {
...
+  copyInputToOutputSchema() {
+     // If there is no information of output schema in the node config then just mantain an output schema for UI purposes.
+     this.state.groupsConfig.outputSchema.isOutputSchemaExists = true;
+     this.state.node.outputSchema = angular.toJson({fields: this.state.node.inputSchema});
+     // FIXME: This is stupid and it is here because of 2-way binding. We bind the model of schema editor to the output schema
+     // The schema editor updates the schema from within and it should expect changes from outside (in this case copy to Output button).
+     // To differntiate editing from within and changes from outside we have to do this.
+     // And we need to differentiate because while editing we don't want to lose focus on the current textbox ($watch on the same property loses it)
+     // One more reason we should have used Flux/Redux so that we could have eliminated these hacks.
+     this.state.outputSchemaUpdate +=1;
```
Comment (elfenheart) [inline cdap-ui/app/features/hydrator/controllers/create/partials/nodeconfig-ctrl.js]: You can use EventPipe instead of doing this?

**Comment by elfenheart** [comment]
I think for Detail View, we should hide the buttons instead of just disabling it.

screen shot 2016-01-12 at 11 32 41 am

File: cdap-ui/app/directives/widget-container/widget-schema-editor/widget-schema-editor.js [hunk 2]
```diff
@@ -215,6 +215,10 @@ angular.module(PKG.name + '.commons')
         }
 
         initialize($scope.model);
+        EventPipe.on('plugin-outputschema.update', function() {
+          console.log('And the model is: ', $scope.model);
```
Comment (elfenheart) [inline cdap-ui/app/directives/widget-container/widget-schema-editor/widget-schema-editor.js]: remove?

File: cdap-ui/app/directives/widget-container/widget-schema-editor/widget-schema-editor.js [hunk 3]
```diff
@@ -215,6 +215,10 @@ angular.module(PKG.name + '.commons')
...
+          initialize($scope.model);
+        });
 
         EventPipe.on('plugin.reset', function () {
           $scope.model = angular.copy(modelCopy);
```
Comment (elfenheart) [inline cdap-ui/app/directives/widget-container/widget-schema-editor/widget-schema-editor.js]: Also need to add an unregister on $destroy

**Comment by elfenheart** [comment]
We should remove the propagate schema button on Sinks. Also I think we should hide Propagate and Copy to Output button in detail view. 

There's also another problem with Stream. During creation, the schema doesn't have ts and headers, but after publish, suddenly it has ts and headers

**Comment by elfenheart** [comment]
:+1: LGTM

**Pull request merged**